felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1147,6 +1147,16 @@ func (m *bpfEndpointManager) onInterfaceUpdate(update *ifaceStateUpdate) {
 		return
 	}
 
+	if update.State == ifacemonitor.StateNotPresent && m.bpfAttachType == apiv3.BPFAttachOptionTCX {
+		// Delete the tcx pins if the interface is gone.
+		// Check if the interface still exists, as we might get events out of order.
+		_, err := m.dp.getIfaceLink(update.Name)
```

Comment:
don't you need to remove it anyway? You got an event that it is not present, but the device is present, so there was a moment where it wasn't present, so it is in fact a different device even if with the same name. That would still break the pin, right? And then [tcx->dev](https://github.com/torvalds/linux/blob/master/kernel/bpf/tcx.c#L213-L215) is likely zero otherwise it would point to an invalid pointer.

So I think you should just delete it here. WDYT?